cmd/atlas/atlas.go

```diff
@@ -40,8 +44,14 @@ func Execute(ctx context.Context) {
 
 // initConfig reads in config file and ENV variables if set.
 func initConfig() {
-	if err := config.Load(); err != nil {
-		log.Fatalf("Error loading config: %v", err)
+	if err := config.LoadAtlasCLIConfig(); err != nil {
```

Comment:
I can understand this back and forth may start becoming frustrating but a lot of this changes are making the logic more complicated than it needs to be which also makes reviewing this PR harder than a normal PR, main point logic is going all over the place so here's a high level of what I'd like to see

You don't need to  try to load the config to then try to copy an old config from the cli, there's no direct dependency and you can decouple this.

To do so `cobra.OnInitialize` takes a list of functions, make copying an old config the first thing you do, an example
```go
source, err := os.Open(src)
if err != nil {
  return // file doesn't esist or can't be processed
  // there's also errors.Is(err, os.ErrNotExist) if you want to actually check for not exists
}
defer source.Close()

destination, err := os.Create(dst)
if err != nil {
  return 0, err
}
defer destination.Close()

if _, err := io.Copy(destination, source); err != nil {
// good idea to try to log the error here
}
```
then `initConfig` can stay as it was, the old file was copied on a previous step and there's no tangling between the operations

Reply:
Updated